tools/shared/src/main/scala/org/scalajs/core/tools/optimizer/Linker.scala

```diff
@@ -224,19 +224,16 @@ final class Linker(semantics: Semantics, outputMode: OutputMode,
           fields += field
 
       // Normal methods
-      case m: MethodDef if m.name.isInstanceOf[Ident] =>
-        if (analyzerInfo.isAnySubclassInstantiated &&
-            analyzerInfo.methodInfos(m.name.name).isReachable) {
-          if (m.body == EmptyTree)
+      case m: MethodDef =>
+        if (analyzerInfo.methodInfos(m.name.name).isReachable) {
+          if (m.name.isInstanceOf[StringLiteral])
+            exportedMembers += linkedMethod(m)
+          else if (m.body == EmptyTree)
             abstractMethods += linkedMethod(m)
           else
             memberMethods += linkedMethod(m)
         }
 
-      case m: MethodDef if m.name.isInstanceOf[StringLiteral] =>
-        if (analyzerInfo.isAnySubclassInstantiated)
-          exportedMembers += linkedMethod(m)
-
```

Comment:
Ah, ah :) this is were we made the exports on normal classes non-aggressive. Your solution is definitely better, but you'll have to adapt the analyzer (where I commented).